created file adding volume component and details of openshift component (#132)
devfile/docs: docs/modules/user-guide/partials/proc_adding-volume-components.adoc

[id="proc_adding-volume_{context}"]
= Adding `volume` components to a devfile

[role="_abstract"]

> Use volume components in order to share files among container components, making it easier to collaborate with other teams during the development process.

Author: I wanted to give a `why` they'd care about sharing files across container components. What do y'all think of this `why I care` statement at the end of this abstract?

Reviewer: This is very good information to provide. It speaks directly to the user story.

> Use volume components in order to share files among container components, making it easier to collaborate with other teams during the development process.

Reviewer: ```suggestion
Use volume components to share files among container components and collaborate with other teams during the development process.
```

.Prerequisites

* xref:adding-schema-version-to-a-devfile.adoc[]
* xref:adding-a-name-to-a-devfile.adoc[]
* xref:adding-components-to-a-devfile.adoc[]

.Procedure

> * Define a component using the type `volume`.

Reviewer: Would this phrasing create a more direct association with the example that follows?
```suggestion
* Define a `volume` component in your devfile.
```

+

> .A minimal `volume` component

Reviewer: Is this example the user's primary source of information about `volume` components? If so, you might want to cover any volume-related attributes in this example or topic, for example, by using callouts. Otherwise, consider providing a cross-reference to the reference topic where this information is available.

Reviewer: Okay to use monospaced fonts in headings?

Author: Dev review: do you have suggestion to add to to this yaml sample to address Rolfe's points above?

[source,yaml]

> ----

Author: Specifically for devs, please verify the accuracy and correctness of this source yaml :)

Reviewer 2: This is correct
